tasks/main.yml

```diff
@@ -1,20 +1,63 @@
 ---
+- name: Update repositories cache and install java
+  apt:
+    name: default-jdk
+    update_cache: yes
+  sudo: yes
+
+- name: Create destination folder
+  file:
+    path: "{{ dest }}"
+    state: directory
+    owner: "{{ server_user }}"
+    group: "{{ server_group }}"
+    mode: 0755
+  sudo: yes
+  sudo_user: "{{ server_user }}"
+
 - name: Download spark-hadoop
   get_url:
     url: https://d3kbcqa49mib13.cloudfront.net/spark-2.1.1-bin-hadoop2.7.tgz
     dest: /tmp/spark-2.1.1-bin-hadoop2.7.tgz
     mode: 0440
+  sudo: yes
+  sudo_user: "{{ server_user }}"
 
-- unarchive:
+- name: Decompress spark-hadoop
+  unarchive:
     src: /tmp/spark-2.1.1-bin-hadoop2.7.tgz
-    dest: /home/tinkerware/
+    dest: "{{ dest }}/"
```

Comment:
you could define here a path name to keep it portable for the future. Like {{ dest }}/spark